src/rpm/common.go

```diff
@@ -0,0 +1,298 @@
+package rpm
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/tarantool/cartridge-cli/src/common"
+)
+
+type rpmValueType int32
+
+type rpmTagType struct {
+	ID    int
+	Type  rpmValueType
+	Value interface{}
+}
+
+type rpmTagSetType []rpmTagType
+
+type packedTagType struct {
+	Count int
+	Data  *bytes.Buffer
+}
+
+func (tagSet *rpmTagSetType) addTags(tags ...rpmTagType) {
+	*tagSet = append(*tagSet, tags...)
+}
+
+func packValues(values ...interface{}) *bytes.Buffer {
+	buf := bytes.NewBuffer(nil)
+
+	for _, v := range values {
+		binary.Write(buf, binary.BigEndian, v)
+	}
+
+	return buf
+}
+
+// All pack tag functions has a link to the corresponding function in
+// the old Lua code.
+// Bless @knazarov for writing this code once,
+// I just rewrote it in Go.
+
+// pack_value
+// https://github.com/tarantool/cartridge-cli/blob/cafd75a5c8ddfdb93ef8290d6e4ebd5d83e4c46e/cartridge-cli.lua#L1982
+func packTag(tag rpmTagType) (*packedTagType, error) {
+	var packed packedTagType
+	packed.Data = bytes.NewBuffer(nil)
+
+	switch tag.Type {
+	case rpmTypeNull: // NULL
+		if tag.Value != nil {
+			return nil, fmt.Errorf("NULL value should be nil")
+		}
+
+		packed.Count = 1
+	case rpmTypeChar: // CHAR
+		fallthrough
+	case rpmTypeBin: // BIN
+		byteArray, ok := tag.Value.([]byte)
+		if !ok {
+			return nil, fmt.Errorf("BIN and CHAR values should be []byte")
+		}
+
+		packed.Count = len(byteArray)
+		for _, byteValue := range byteArray {
+			if _, err := io.Copy(packed.Data, packValues(byteValue)); err != nil {
+				return nil, err
+			}
+		}
+	case rpmTypeStringArray: // STRING_ARRAY
+		// value should be strings array
+		stringsArray, ok := tag.Value.([]string)
+		if !ok {
+			return nil, fmt.Errorf("STRING_ARRAY value should be []string")
+		}
+
+		packed.Count = len(stringsArray)
+
+		for _, v := range stringsArray {
+			bytedString := []byte(v)
+			bytedString = append(bytedString, 0)
+			if _, err := io.Copy(packed.Data, packValues(bytedString)); err != nil {
+				return nil, err
+			}
+		}
+	case rpmTypeString: // STRING
+		// value should be string
+		stringValue, ok := tag.Value.(string)
+		if !ok {
+			return nil, fmt.Errorf("STRING value should be string")
+		}
+
+		packed.Count = 1
+
+		bytedString := []byte(stringValue)
+		bytedString = append(bytedString, 0)
+		if _, err := io.Copy(packed.Data, packValues(bytedString)); err != nil {
+			return nil, err
+		}
+
```

Comment:
Since this line you change code style and add an empty line before each following "case".
Could you do it for the lines above? Or remove it below?